### DISCUSSION TITLE
0 Ohm resistor

### DISCUSSION THREAD

**Discussion by rede676** [SpiceSharp/SpiceSharp#216] [General]
I saw in the source that it says the following:

```
        /// <remarks>
        /// If the resistance is limited to <see cref="MinimumResistance" /> to avoid numerical instability issues. 
        /// If a 0 Ohm resistance is wanted, consider using an ideal voltage source instead.
        /// </remarks>
```
`ideal voltage source` isn't a component, is there something I am missing, What is the best way to achieve a 0 Ohm resistor?

**Comment by svenboulanger** [comment]
The `VoltageSource` component represents an idealized voltage source. You can give it a value of 0V to act like a short between two nodes.

**Reply by rede676** [reply]
Thank you, so if I wanted to put a 0 ohm resister across the output of a voltage divider, It would look like this:
```
var ckt = new Circuit(
    new VoltageSource("V1", "in", "0", 5),
    new Resistor("R1", "in", "middle1", 100),
    new Resistor("R2", "middle2", "0", 100),
    new VoltageSource("0Ohm", "middle1", "middle2",  0) //0 Ohm resister
);
```
Would I need to add the VoltageSource to the sweep?

**Reply by svenboulanger** [reply]
That's indeed the way to do it. The voltage source of 0V simply represents a short-circuit between two nodes, there's no reason to sweep it.

If you want to sweep a resistor, consider not sweeping from 0 but from a very small value, e.g. `1e-6` (1 micro-ohm) or even `1e-9`. While the result will be slightly off, it will usually be more than close enough. Then you don't have to deal with adding an extra voltage source in your divider. This is also how conventional SPICE simulators will do it (maybe without explicitly telling you).
